common-template/src/main/java/com/redhat/cloud/notifications/qute/templates/TemplateService.java

```diff
@@ -1,52 +1,111 @@
 package com.redhat.cloud.notifications.qute.templates;
 
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
 import com.redhat.cloud.notifications.ingress.Action;
+import com.redhat.cloud.notifications.qute.templates.mapping.AnsibleAutomationPlatform;
 import com.redhat.cloud.notifications.qute.templates.mapping.Console;
+import com.redhat.cloud.notifications.qute.templates.mapping.DefaultInstantEmailTemplates;
 import com.redhat.cloud.notifications.qute.templates.mapping.DefaultTemplates;
 import com.redhat.cloud.notifications.qute.templates.mapping.OpenShift;
 import com.redhat.cloud.notifications.qute.templates.mapping.Rhel;
+import com.redhat.cloud.notifications.qute.templates.mapping.SecureEmailTemplates;
 import com.redhat.cloud.notifications.qute.templates.mapping.SubscriptionServices;
 import io.quarkus.logging.Log;
 import io.quarkus.qute.Engine;
 import io.quarkus.qute.TemplateInstance;
 import io.quarkus.runtime.Startup;
 import jakarta.annotation.PostConstruct;
 import jakarta.enterprise.context.ApplicationScoped;
-import jakarta.inject.Inject;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import java.io.File;
 import java.util.HashMap;
 import java.util.Map;
 
 @Startup
 @ApplicationScoped
 public class TemplateService {
 
-    @Inject
-    Engine engine;
+    private static final String SECURED_EMAIL_TEMPLATES = "notifications.use-secured-email-templates.enabled";
+    private static final String DEFAULT_TEMPLATE = "notifications.use-default-template";
 
+    // Only used in special environments.
+    @ConfigProperty(name = SECURED_EMAIL_TEMPLATES, defaultValue = "false")
+    boolean useSecuredEmailTemplates;
+
+    // Only used in stage environments.
+    @ConfigProperty(name = DEFAULT_TEMPLATE, defaultValue = "false")
+    boolean defaultEmailTemplateEnabled;
+
+    final Engine engine;
+
+    final ObjectMapper objectMapper;
+
+    String emailTemplatePrefix = "Secure/";
     Map<TemplateDefinition, String> templatesConfigMap = new HashMap<>();
 
+    public TemplateService(Engine engine, ObjectMapper objectMapper) {
+        this.engine = engine;
+        this.objectMapper = objectMapper;
+    }
+
+    public String getTemplatePrefix(final IntegrationType integrationType) {
+        if (isSecuredEmailTemplatesEnabled() &&
+            integrationType.getRootFolder().equals(IntegrationType.EMAIL_BODY.getRootFolder())) {
+            return emailTemplatePrefix;
+        } else {
+            return "";
+        }
+    }
```

Comment:
Is there a reason or preference to include the logic here, instead of just adding a constant to `SecureEmailTemplates` and prepending it there? Example

```java
public class SecureEmailTemplates {
    static final String SECURE_FOLDER_NAME = "Secure/";
    
    public static final Map<TemplateDefinition, String> templatesMap = Map.ofEntries(
            entry(new TemplateDefinition(EMAIL_DAILY_DIGEST_BODY, BUNDLE_NAME, ADVISOR_APP_NAME, null), SECURE_FOLDER_NAME + ADVISOR_FOLDER_NAME + "dailyEmailBody.html"),
            // [trimmed]
```